demos/text_detection_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,509 @@
+        if (tdRequired) {
+            // Text detection
+            cv::GMat inDet, det1, det2;
+            if (1 == tdInputChannels) {
+                inDet = cv::gapi::BGR2Gray(in);
+            } else {
+                inDet = in;
+            }
+            std::tie(det1, det2) = cv::gapi::infer<nets::TextDetection>(inDet);
+            // ROI for each text piece
+            rrs = custom::DetectionPostProcess::on(det1, det2, size, tdInputSize,
+                                                   segmConfThreshold, linkConfThreshold,
+                                                   tdMaxRectsNum);
+        } else {
+            rrs = cv::GArray<cv::RotatedRect>(emptyBox);
+        }
+        cv::GArray<std::vector<cv::Point2f>> pts;
+        if (trRequired) {
+            cv::GMat inRec;
+            if (1 == trInputChannels) {
+                inRec = cv::gapi::BGR2Gray(in);
+            } else {
+                inRec = in;
+            }
+            // Labels preprocessed for recognition for each ROI
+            cv::GArray<cv::GMat> labels;
+            std::tie(labels, pts) = custom::CropLabels::on(inRec, rrs, trInputDims, centralCrop);
+            // Text Recognition
+            cv::GArray<cv::GMat> texts;
+            if (trComposite) {
+                cv::GArray<cv::GMat> hiddens;
+                cv::GArray<cv::GMat> features;
+                std::tie(hiddens, features) =
+                    cv::gapi::infer2<nets::TextRecognitionEncoding>(inRec, labels);
+                texts = custom::CompositeTRDecode::on(hiddens, features, decNetwork,
+                                                      decoderNumClasses, decoderEndToken);
+            } else {
+                texts = cv::gapi::infer2<nets::TextRecognition>(inRec, labels);
+            }
+            outs += cv::GOut(texts);
+        } else {
+            pts = custom::PointsFromRRects::on(rrs, size, centralCrop);
+        }
+        outs += cv::GOut(pts);
+        // Inputs and outputs of graph
+        cv::GComputation graph(cv::GIn(in), std::move(outs));
+        /** ---------------- End of graph ---------------- **/
+        auto kernels  = custom::kernels();
+        auto pipeline = graph.compileStreaming(cv::compile_args(kernels, networks));
+
+        // Output containers for results
+        cv::Mat image;
+        std::vector<std::vector<cv::Point2f>> outPts;
+        std::vector<cv::Mat> outTexts;
+        auto outVector = cv::gout(image);
+        if (trRequired) {
+            outVector += cv::gout(outTexts);
+        }
+        outVector += cv::gout(outPts);
+        /** ---------------- The execution part ---------------- **/
+        pipeline.setSource(cv::gin(cv::gapi::wip::make_src<custom::CommonCapSrc>(cap)));
+        pipeline.start();
+        std::chrono::steady_clock::time_point beginFrame = std::chrono::steady_clock::now();
+        while (pipeline.pull(std::move(outVector))) {
+            cv::Mat demoImage = image.clone();
```

Comment:
Why clone?

Reply:
`demoImage` is used for the user output, so I'm cloning not to corrupt the original frame, just in case

Reply:
removed